tests/libcxx/enc/CMakeLists.txt

```diff
@@ -83,6 +85,10 @@ function(add_libcxx_test_enc NAME CXXFILE)
     enclave_compile_definitions(libcxxtest-${NAME}_enc PRIVATE -DWITH_MAIN -D__TEST__="${CXXFILE}")
     enclave_link_libraries(libcxxtest-${NAME}_enc libcxxtest-support)
 
+    if (CXXFILE IN_LIST cxx17tests)
+        set_enclave_property(TARGET libcxxtest-${NAME}_enc PROPERTY CXX_STANDARD 17)
```

Comment:
Is this the only place we need to set the compiler option? I would expect that if we claim to support C++17 that we should set it globally for the OE SDK as part of cmake/compiler_settings.cmake ... maybe this could also result in the gcc side failures since oelibcxx is being compiled with C++14 standard still.

Reply:
Update from my experiments
- Applying C++17 globally works both for clang and gcc (all default libcxx tests passed)
- Some tests (full libcxx tests) failed with C++17. I can compile a list of such tests and see how many are they.

Reply:
Update: There are at least 73 cases that are not supported by C++17. This means that we always need to keep such a list regardless using C++14 or updating to C++17. However, the list of C++17-dependent (291 cases) does less than C++14-dependent one.

Reply:
@mingweishih Can you please help me understand this statement
> However, the list of C++17-dependent (291 cases) does less than C++14-dependent one.
Are there more tests for C++14 than there are for C++17?

Reply:
> @mingweishih Can you please help me understand this statement.Are there more tests for C++14 than there are for C++17?
> 
> > However, the list of C++17-dependent (291 cases) does less than C++14-dependent one.

Ah, my mistake. The statement should be "However, the list of C++17-dependent (291 cases) does more than C++14-dependent one." instead.

Reply:
Yes, some test cases (updated number is 58) I mentioned are failing with c++17, but passing with c++14. The requirement of each test is well-documented in the comments of each test.

Those failing tests mostly depend on the API changes introduced in c++17.
For example, binary_function was removed in c++17:
https://github.com/openenclave/openenclave/blob/ee6782c3253fe941084185cb215ff48a122a6104/3rdparty/libcxx/libcxx/test/std/depr/depr.function.objects/depr.base/binary_function.pass.cpp#L12

Similarly, unary_function was also removed in c++17:
https://github.com/openenclave/openenclave/blob/ee6782c3253fe941084185cb215ff48a122a6104/3rdparty/libcxx/libcxx/test/std/depr/depr.function.objects/depr.adaptors/depr.member.pointer.adaptors/mem_fun1_t.pass.cpp#L11

It turns out that there are only two such cases between c++11 and c++14.
https://github.com/openenclave/openenclave/blob/ee6782c3253fe941084185cb215ff48a122a6104/3rdparty/libcxx/libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/sized_delete_array11.pass.cpp#L15
and
https://github.com/openenclave/openenclave/blob/ee6782c3253fe941084185cb215ff48a122a6104/3rdparty/libcxx/libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.single/sized_delete11.pass.cpp#L15

They are currently classified as unsupported.
https://github.com/openenclave/openenclave/blob/ee6782c3253fe941084185cb215ff48a122a6104/tests/libcxx/tests.unsupported#L298
and
https://github.com/openenclave/openenclave/blob/ee6782c3253fe941084185cb215ff48a122a6104/tests/libcxx/tests.unsupported#L290